consensus/metrics.go

```diff
@@ -50,101 +54,107 @@ type Metrics struct {
 }
 
 // PrometheusMetrics returns Metrics build using Prometheus client library.
-func PrometheusMetrics(namespace string) *Metrics {
+// Optionally, labels can be provided along with their values ("foo",
+// "fooValue").
+func PrometheusMetrics(namespace string, labelValues ...string) *Metrics {
+	labels := []string{}
+	for i := 0; i < len(labelValues)/2; i += 2 {
+		labels = append(labels, labelValues[i])
+	}
 	return &Metrics{
 		Height: prometheus.NewGaugeFrom(stdprometheus.GaugeOpts{
 			Namespace: namespace,
 			Subsystem: MetricsSubsystem,
 			Name:      "height",
 			Help:      "Height of the chain.",
-		}, []string{}),
+		}, labels).With(labelValues...),
```

Comment:
How do these labels work? We pass in just the values as an argument to NewGaugeFrom and then the whole list of keys and values into .With?

Reply:
Yes. See the example here https://github.com/go-kit/kit/blob/1e404df823eb187378b081f22f00e4e38f05a8bd/metrics/prometheus/prometheus_test.go#L32-L37